File: main/src/main/java/org/anibyl/slounik/dialogs/AboutDialog.java
package org.anibyl.slounik.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.widget.TextView;
import org.anibyl.slounik.R;
import org.anibyl.slounik.Util;

/**
 * About dialog.
 *
 * Created by Usievaład Čorny on 05.04.2015 9:36.
 */
public class AboutDialog extends Dialog {
    private final String title;
    private final Context context;

    public AboutDialog(Context context, String title) {
        super(context);

        this.title = title;
        this.context = context;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.about);
        setTitle(title);
        TextView androidIdField = (TextView) findViewById(R.id.about_android_id);
        androidIdField.setText(androidIdField.getText().toString().replace("%s", Util.getAndroidId(context)));
    }
}

File: main/src/main/kotlin/org/anibyl/slounik/dialogs/AboutDialog.kt
package org.anibyl.slounik.dialogs

import android.app.Dialog
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import org.anibyl.slounik.BuildConfig
import org.anibyl.slounik.R


/**
 * About information dialog.
 *
 * @author Sieva Kimajeŭ
 * @created 2016-12-26
 */
class AboutDialog : DialogFragment() {
	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
		val view: View = inflater.inflate(R.layout.about, container, false)

		val version: TextView = view.findViewById(R.id.about_dialog_version)
		val homepageButton = view.findViewById(R.id.about_dialog_homepage_button) as Button
		val betaParticipationButton = view.findViewById(R.id.about_dialog_beta_participation_button) as Button
		val closeButton = view.findViewById(R.id.about_dialog_close_button) as Button

		version.text = BuildConfig.VERSION_NAME

		homepageButton.setOnClickListener {
			startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(resources.getString(R.string.homepage))))
		}

		betaParticipationButton.setOnClickListener {
			startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(resources.getString(R.string.beta_participation_page))))
		}

		closeButton.setOnClickListener {
			dismiss()
		}

		isCancelable = true

		return view
	}

	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
		val dialog = super.onCreateDialog(savedInstanceState)

		dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)

		return dialog
	}
}
